File: arreglos 27-08-2025/arreglos.js

let frutas = [];

// Un arreglo con elementos iniciales
let numeros = [1, 2, 3, 4, 5];
console.log("Arreglo inicial de números:", numeros);

// 2. Acceder a elementos
// Los índices de los arreglos comienzan en 0
let primerNumero = numeros[0]; // Accede al primer elemento (1)
console.log("El primer número es:", primerNumero);

// 3. Modificar elementos
numeros[2] = 10; // Cambia el tercer elemento (3) a 10
console.log("Arreglo modificado:", numeros);

// 4. Añadir elementos
// .push() añade un elemento al final del arreglo
numeros.push(6);
console.log("Arreglo después de push(6):", numeros);

// 5. Longitud del arreglo
console.log("La longitud del arreglo es:", numeros.length);

// 6. Iterar sobre un arreglo
console.log("Iterando con un bucle for:");
for (let i = 0; i < numeros.length; i++) {
  console.log(`Elemento en el índice ${i}: ${numeros[i]}`);
}

console.log("Iterando con forEach:");
numeros.forEach(function(numero, indice) {
  console.log(`Elemento en el índice ${indice}: ${numero}`);
});

File: arreglos 27-08-2025/arreglos.py
# En Python, el tipo de dato más similar a un arreglo es la 'lista' (list).
# Las listas son ordenadas, mutables (se pueden cambiar) y pueden contener elementos de diferentes tipos.

# 1. Declaración e inicialización
# Una lista vacía
colores = []

# Una lista con elementos iniciales
numeros = [1, 2, 3, 4, 5]
print(f"Lista inicial de números: {numeros}")

# 2. Acceder a elementos
# Los índices comienzan en 0
primer_numero = numeros[0]  # Accede al primer elemento (1)
print(f"El primer número es: {primer_numero}")

# 3. Modificar elementos
numeros[2] = 10  # Cambia el tercer elemento (3) a 10
print(f"Lista modificada: {numeros}")

# 4. Añadir elementos
# .append() añade un elemento al final de la lista
numeros.append(6)
print(f"Lista después de append(6): {numeros}")

# 5. Longitud de la lista
print(f"La longitud de la lista es: {len(numeros)}")

# 6. Iterar sobre una lista
print("Iterando con un bucle for:")
for numero in numeros:
    print(f"- {numero}")

